### PR TITLE
Fix cross module node globals

### PR DESCRIPTION
**Pull request by masad-frost**
## Bug

When running with an entrypoint, the entry point's globals/context is different from required modules. This breaks some packages such as math.js that rely on instanceof. You can see the bug in the newly added asserts https://github.com/replit/prybar/commit/10c16f9e064d0ec3f522df76276c1f4f5eaf9b2e.

## Fix

Use the (current) global context when running the file, i.e. `script.runInThisContext`. This change gives us the piece of mind going forward when dealing with the global context. We still patch somethings, like adding `alert`, `confirm`, and `prompt`. We also patch the `global.module` and `global.require` to make sure the context thinks we're in the entrypoint file rather than the `repl.js` file.

I also fixed a bunch of tests that were broken.

### CODE REVIEW

**Comment by masad-frost** [comment]
https://app.asana.com/0/inbox/974357908413931/1198514514357824/1199218400429100

File: test_files/module_patching.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+console.assert(module.id === '.')
```
Comment (sergeichestakov) [inline test_files/module_patching.js:1]: just curious: why do we `console.assert` instead of using Node's built in assertion library?

**Review by sergeichestakov — APPROVED** [review]
Sweet! Thanks for the 🔨

File: test_files/module_patching.js (hunk 1 above)
Comment (masad-frost) [reply test_files/module_patching.js:1]: Either way works, we're just comparing outputs from running this script. I just went with what's already there.

Thanks for review. The bigger challenge now is moving node to polygott

**Pull request merged**